### ISSUE TITLE
 Affinity propagation did not converge

### ISSUE THREAD

**Issue by yuezhiTang** [medema-group/BiG-SCAPE#47]
When I used Bigscape to process tasks, my data was approximately 50000 BGCs,the command line I am using is (nihup python3 /path/BiG-SCAPE-1.1.5/bigscape.py -i /path/output -o /path/res --mode auto --mibig &), and the following error occurred, which I cannot understand and resolve:

Working for each BGC class
  Sorting the input BGCs

  PKSI (14928 BGCs)
   Writing annotation files
   Calculating all pairwise distances
Ignored unknown character X (seen 2 times)
Ignored unknown character X (seen 1 times)
Ignored unknown character X (seen 3 times)
Ignored unknown character X (seen 1 times)
Ignored unknown character X (seen 1 times)
Ignored unknown character X (seen 1 times)
Ignored unknown character X (seen 3 times)
Ignored unknown character X (seen 1 times)
Ignored unknown character X (seen 4 times)
Ignored unknown character X (seen 4 times)
Ignored unknown character X (seen 1 times)
Ignored unknown character X (seen 1 times)
Ignored unknown character X (seen 4 times)
Ignored unknown character X (seen 1 times)
Ignored unknown character X (seen 1 times)
Ignored unknown character X (seen 1 times)
/home/lvhuawei/miniconda3/envs/bigscape/lib/python3.11/site-packages/sklearn/cluster/_affinity_propagation.py:142: ConvergenceWarning: Affinity propagation did not converge, this model may return degenerate cluster centers and labels.
  warnings.warn(

**Comment by adraismawur** [comment]
Hi!

This is a warning from scikit-learn telling you that the affinity propagation could not settle on clusters and labels (it never converges to consistent clusters).
This should just remain a warning, but some versions of scikit-learn end up returning invalid cluster centers and labels. 

Does the rest of the process still complete? And which version of Scikit-learn are you using?

**Comment by yuezhiTang** [comment]
> Hi!
> 
> This is a warning from scikit-learn telling you that the affinity propagation could not settle on clusters and labels (it never converges to consistent clusters). This should just remain a warning, but some versions of scikit-learn end up returning invalid cluster centers and labels.
> 
> Does the rest of the process still complete? And which version of Scikit-learn are you using?

Thank you for your reply. When my nohup.out file reports this information, bigscape has already stopped running and will not continue. I have tried deleting the file or giving a new output directory to restart bigscape, but it always stops at this step. The version of scikit-learn I am using is 1.3.0. Additionally, I would like to ask: how is the visualization done for the results obtained from other smaller datasets that I have successfully completed with bigscape? When I open the xx.html webpage file generated by the bigscape task, it seems that the images mentioned in the literature are not present. The results processed with other visualization software do not meet my expectations.

**Comment by adraismawur** [comment]
Hello,

Could you try using scikit-learn version 0.19.2?

Regarding the visualizations, what images are you referring to exactly? BiG-SCAPE generates image files under [output_folder]/SVG for each BGC.

**Comment by CatarinaCarolina** [comment]
Hi!

So an svg from each GCF tree can be downloaded from its GCF page on the html. it is generated 'there on the fly'.
(svgs from each BGC can be found in the [output_folder]/SVG for each BGC).
The BiG-SCAPE CORASON publication also features BGC trees generated by CORASON, perhaps that is also what you are looking for?

Alternatively if you are referring to the network clustering, BiG-SCAPE doesnt produce svgs for this but you will want to load the network file ([output_folder]/network_files/run_id/class_id/file.network) into a tool such as cytoscape to visualise, edit and export the network in a image file format of your choice.
